### ISSUE TITLE
Vision Module in "Stale" Mode Due to Mismatched Monitor IDs

### ISSUE THREAD

**Issue by meltiso** [mediar-ai/screenpipe#369]
#### Description:
At app launch, the Vision module enters "stale" mode with the status `frame: stale, audio: ok`. Upon checking the logs, the following errors appear:

```
2024-09-27T10:47:16.683910Z ERROR screenpipe_vision::core: Failed to get monitor with id: 330083. Exiting continuous_capture.    
2024-09-27T10:47:16.683956Z ERROR screenpipe_vision::core: Failed to get monitor with id: 20580179. Exiting continuous_capture. 
```

In the app settings under the monitors section, I see:

```
79825613. \\.\DISPLAY1 (default) - 1920x1080
63768545. \\.\DISPLAY2 - 1920x1080
```

The monitor IDs in the logs don't match the ones detected in the settings, which seems to be causing the Vision module to go stale, as it can't use the detected monitors.

#### Steps to Reproduce:
1. Launch the Screenpipe app on Windows 11.
2. Close Screenpipe completely
3. Disconnect and reconnect an HDMI cable for one of the monitors (I don't know if this step matters).
4. Relaunch Screenpipe.
5. Go to **settings** -> **monitors**, and notice one of the monitor IDs have changed.
6. Click **update** to restart the Vision and Audio modules.
7. Open the logs and observe errors related to monitor IDs.
8. Check that the Vision module is in "stale" mode, failing on both screens due to mismatched monitor IDs.

#### Expected Behavior:
- The Vision module should be able to capture frames using the detected monitors.
- The monitor IDs in the logs should match the IDs of the detected screens, allowing the Vision module to function correctly.

#### Actual Behavior:
- The Vision module enters "stale" mode, failing to capture frames.
- The logs show errors for monitor IDs that do not exist in the list of detected displays.

#### Environment:
- **App Version**: 0.2.72
- **OS**: Windows 11
- **Monitor Configuration**: Dual monitors, both 1920x1080 resolution. One in HDMI.

**Comment by meltiso** [comment]
I realized that even though only two screens show up in the list, Screenpipe is counting a total of four screens. I didn’t select the two screens I mentioned in my original issue, which I just noticed.

After selecting those two screens, the total reaches four, and the Vision module status changes to `ok`. However, we still see errors in the logs related to the screens that aren’t in the list but are included in the total count. 
![{A49EE733-8860-4995-BDEB-53CD2822185A}](https://github.com/user-attachments/assets/8040363f-eeb6-45f5-947a-54c8737a3ff3)
![{B947CDD5-B9D1-46D7-B856-CA13A1A0CA65}](https://github.com/user-attachments/assets/4ed8baf7-9a27-4fc2-af78-636912963b81)
